tss-esapi/src/structures/tagged/signature.rs

```diff
@@ -125,3 +128,59 @@ impl TryFrom<TPMT_SIGNATURE> for Signature {
         }
     }
 }
+
+impl Marshall for Signature {
+    const BUFFER_SIZE: usize = std::mem::size_of::<TPMT_SIGNATURE>();
+
+    fn marshall(&self) -> Result<Vec<u8>> {
+        let tpmt_sig = TPMT_SIGNATURE::try_from(self.clone())?;
+        let mut offset = 0;
+        let mut buffer = Vec::with_capacity(Self::BUFFER_SIZE);
+
+        let ret = unsafe {
+            Tss2_MU_TPMT_SIGNATURE_Marshal(
+                &tpmt_sig,
+                buffer.as_mut_ptr(),
+                buffer.capacity().try_into().map_err(|e| {
+                    error!("Failed to convert size of buffer to TSS size_t type: {}", e);
+                    Error::local_error(WrapperErrorKind::InvalidParam)
+                })?,
+                &mut offset,
+            )
+        };
+        let ret = Error::from_tss_rc(ret);
+        if ret.is_success() {
+            unsafe {
+                buffer.set_len(offset as usize);
```

Comment:
This should be changed to the vector version that does not ned to use 'unsafe'.

Reply:
Nice, thanks for pointing that out!